crates/proto/src/serialize/txt/rdata_parsers/dnskey.rs

```diff
@@ -0,0 +1,138 @@
+use crate::rr::dnssec::rdata::dnskey::DNSKEY;
+use crate::rr::dnssec::Algorithm;
+use crate::serialize::txt::{ParseError, ParseErrorKind, ParseResult};
+
+fn bit_is_set(value: u16, bit: usize) -> bool {
+    let mask = 1 << bit;
+    value & mask == mask
+}
+
+#[allow(deprecated)]
+pub(crate) fn parse<'i, I: Iterator<Item = &'i str>>(mut tokens: I) -> ParseResult<DNSKEY> {
+    let flags_str: &str = tokens
+        .next()
+        .ok_or_else(|| ParseError::from(ParseErrorKind::Message("flags not present")))?;
+    let protocol_str: &str = tokens
+        .next()
+        .ok_or_else(|| ParseError::from(ParseErrorKind::Message("protocol not present")))?;
+    let algorithm_str: &str = tokens
+        .next()
+        .ok_or_else(|| ParseError::from(ParseErrorKind::Message("algorithm not present")))?;
+
+    let flags: u16 = flags_str.parse()?;
+    if ![0, 256, 257].contains(&flags) {
+        return Err(ParseError::from(ParseErrorKind::Message(
+            "flags field must be one of: 0, 256, 257",
+        )));
+    }
+
+    let secure_entry_point = bit_is_set(flags, 0);
+    let revoke = bit_is_set(flags, 7);
+    let zone_key = bit_is_set(flags, 8);
+
+    let protocol: u8 = protocol_str.parse()?;
+
+    if protocol != 3 {
+        return Err(ParseError::from(ParseErrorKind::Message(
+            "protocol field must be 3",
+        )));
+    }
+
+    let algorithm = Algorithm::from_u8(algorithm_str.parse()?);
+
+    let public_key_str: String = tokens.collect();
+    if public_key_str.is_empty() {
+        return Err(ParseError::from(ParseErrorKind::Message(
+            "public key not present",
+        )));
+    }
+
+    let public_key = data_encoding::BASE64.decode(public_key_str.as_bytes())?;
```

Comment:
Nit: inline this in the `Self` initializer?

Reply:
I kind of like the symmetry of having all arguments to `new` be variables. inlining this one would break the symmetry